conformance/lang/expressions/type_cast_expression.balt

```diff
@@ -0,0 +1,2012 @@
+type MyUnion int|MyError;
+
+function errorFunction() {
+    int|error a = 1;
+    int b = <int>a; // @error attempt to cast away an error
+
+    string c = <string>getString(); // @error attempt to cast away an error
+
+    MyUnion d = 3;
+    int e = <int>d; // @error attempt to cast away an error
+}
+
+function getString() returns error|string {
+    return "car";
+}
+
+Test-Case: output
+Description: Test if static type of expression contains error, then the intersection of T and error must 
+             also be non-empty.
+Labels: error-type, error-constructor-expr, error-type-parameter, field-access-expr, method-call-expr, module-type-defn, 
+        record-type, string, type-cast-expr, union-type
+
+type MyError error<record {|int severity;|}>;
+
+type MyUnion any|MyError;
+
+function init() {
+    MyUnion a = 3;
+    int|error b = <int|error>a;
+    io:println(b); // @output 3
+
+    string|error c = error("msg", severity = 1);
+    MyError e = <MyError> c;
+    io:println(e.detail().severity); // @output 1
+}
+
+Test-Case: output
+Description: Test the contextually expected type for expression is the intersection of the contextually expected type 
+             of the type-cast-expr and the type described by the type-descriptor
+Labels: any, array-type, byte, decimal, DecimalFloatingPointNumber, float, mapping-constructor-expr, 
+        record-rest-descriptor, record-type, record-type-readonly-field, string, tuple-type, type-cast-expr, union-type,
+        value:toBalString
+
+function init() {
+    record {|int a;string...; |} a = <record {|int a;|}>{a: 5};
+    io:println(typeof a); // @output typedesc record {| int a; |}
+
+    record {|int a;string...; |} b = <record {|readonly int a;|}>{a: 5};
+    io:println(typeof b); // @output typedesc record {| readonly int a; |}
+
+    record {|int a;string...; |} c = <record {|readonly int a; "Car" b;|}>{a: 5, b: "Car"};
+    io:println(typeof c); // @output typedesc record {| readonly int a; "Car" b; |}
+
+    record {|int a;string...; |}|string d = <record {|readonly int a; "Car" b;|}|string>{a: 5, b: "Car"};
+    io:println(typeof d); // @output typedesc record {| readonly int a; "Car" b; |}
+
+    int[] e = <[int, byte]>[444, 21];
+    io:println(typeof e); // @output typedesc [int,byte]
+
+    int[]|byte[] f = <[int, byte]>[444, 21];
+    io:println(typeof f); // @output typedesc [int,byte]
+
+    any g = <float>1;
+    io:println(g); // @output 1.0
+
+    any h = <decimal>1.4;
+    io:println(h.toBalString()); // @output 1.4d
+}
+
+Test-Case: output
+Description: Test cast to readonly can be used to cause constructors withing the expression to construct values
+             with the read-only bit on.
+Labels: any, array-type, list-constructor-expr, map-type, mapping-constructor-expr, readonly-type, string, 
+        table-constructor-expr, table-type, type-cast-expr, typeof-expr
+
+function init() {
+    int[] a = <readonly>[1, 2];
+    io:println(typeof a); // @output typedesc [1,2]
```

Comment:
IMO using an is check is better since it won't depend on toString of typedesc. 

e.g., `a is int[2] & readonly` and/or `a is [1, 2] & readonly`